tests/robustness/traffic/etcd.go

```diff
@@ -105,87 +110,120 @@ func (t etcdTraffic) Run(ctx context.Context, c *RecordingClient, limiter *rate.
 		}
 		key := fmt.Sprintf("%d", rand.Int()%t.keyCount)
 		// Avoid multiple failed writes in a row
-		if !lastOperationSucceeded {
-			_, err := t.Read(ctx, c, key)
-			if err != nil {
-				continue
-			}
-			limiter.Wait(ctx)
+		if lastOperationSucceeded {
+			requestType = pickRandom(t.requests)
+		} else {
+			requestType = Get
 		}
-		err := t.RandomOperation(ctx, c, limiter, key, ids, lm)
+		rev, err := t.Request(ctx, c, requestType, limiter, key, ids, lm, lastRev)
 		lastOperationSucceeded = err == nil
 		if err != nil {
 			continue
 		}
+		if rev != 0 {
+			lastRev = rev
+		}
 		limiter.Wait(ctx)
 	}
 }
 
-func (t etcdTraffic) Read(ctx context.Context, c *RecordingClient, key string) (*mvccpb.KeyValue, error) {
-	getCtx, cancel := context.WithTimeout(ctx, RequestTimeout)
-	resp, err := c.Get(getCtx, key)
-	cancel()
-	return resp, err
-}
-
-func (t etcdTraffic) RandomOperation(ctx context.Context, c *RecordingClient, limiter *rate.Limiter, key string, id identity.Provider, lm identity.LeaseIdStorage) error {
+func (t etcdTraffic) Request(ctx context.Context, c *RecordingClient, request etcdRequestType, limiter *rate.Limiter, key string, id identity.Provider, lm identity.LeaseIdStorage, lastRev int64) (rev int64, err error) {
 	opCtx, cancel := context.WithTimeout(ctx, RequestTimeout)
 
-	var err error
-	switch pickRandom(t.operations) {
+	switch request {
+	case StaleGet:
+		_, rev, err = c.Get(opCtx, key, lastRev)
```

Comment:
No sure that I understand it correctly: The `lastRev` is from last command. It seems like the `lastRev` is always latest one. Is it still considered as stale?

Reply:
It's the latest one for the client, but with multiple clients it's always a little stale 1-5 revisions behind the latest request.